python-package/xgboost/dask/__init__.py

```diff
@@ -1517,15 +1530,73 @@ def predict(
         base_margin: Optional[_DaskCollection] = None,
         iteration_range: Optional[IterationRange] = None,
     ) -> Any:
-        return self.client.sync(
-            self._predict_async,
-            X,
+        iteration_range = self._get_iteration_range(iteration_range)
+        booster_options = dict(
+            base_margin=base_margin,
+            missing=self.missing,
+        )
+        predict_options = dict(
             output_margin=output_margin,
             validate_features=validate_features,
-            base_margin=base_margin,
             iteration_range=iteration_range,
         )
 
+        is_regression = isinstance(self, XGBRegressorBase)
+
+        meta: numpy.typing.NDArray[numpy.float32]
```

Comment:
`meta` here could be a cupy.ndarray too. Haven't looked into how to handle that yet.

Reply:
The return type of `Booster.predict` is `np.ndarray`, so using `np.ndarray` here matches that.